app/src/main/java/com/google/android/soundchecker/mediacodec/AudioDecoderSource.kt

```diff
@@ -0,0 +1,252 @@
+/**
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.android.soundchecker.mediacodec
+
+import android.media.AudioFormat
+import android.media.MediaCodec
+import android.media.MediaCodecList
+import android.media.MediaFormat
+import android.util.Log
+import com.google.android.soundchecker.utils.AudioSource
+import java.nio.ByteBuffer
+import java.nio.charset.Charset
+import java.util.Arrays
+import kotlin.math.min
+
+class AudioDecoderSource() : AudioSource() {
+
+    companion object {
```

Comment:
This needs to be at the bottom of the class,

https://kotlinlang.org/docs/coding-conventions.html#class-layout

Reply:
Thanks!